### PR TITLE
RAB-1294: [Front] Display number of attributes in the grid

### PR DESCRIPTION
**Pull request by MadeWilson**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx [hunk 1]
```diff
@@ -16,14 +18,30 @@ const ModalContent = styled.div`
...
-  const [numberOfAttribute] = useState<number>(0);
+  const [numberOfAttribute, setNumberOfAttribute] = useState<number>(0);
+  const [replacementAttributeGroup, setReplacementAttributeGroup] = useState<string | null>(null);
   const inputRef = useRef<HTMLInputElement>(null);
+  const userContext = useUserContext();
 
   useAutoFocus(inputRef);
 
+  useEffect(() => {
+    if (null !== selectedAttributeGroups) {
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:36]: selectedAttributeGroups cannot be null no ?

**Review by pierrejolly — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx [hunk 2]
```diff
@@ -16,14 +18,30 @@ const ModalContent = styled.div`
...
+      let numberOfAttribute = 0;
+      selectedAttributeGroups.forEach(attributeGroup => {
+        numberOfAttribute += attributeGroup.attribute_count;
+      });
+      setNumberOfAttribute(numberOfAttribute);
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:41]:
```suggestion
    setNumberOfAttribute(selectedAttributeGroups.reduce((accumulator, attributeGroup) => accumulator + attributeGroup.attribute_count));
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx [hunk 3]
```diff
@@ -33,7 +51,7 @@ const ModalMassDeleteAttributeGroups = ({attributeGroups, onConfirm}: MassDelete
       <Button level="danger" onClick={() => openMassDeleteModal()}>
         {translate('pim_enrich.entity.attribute_group.mass_delete.button')}
       </Button>
-      {isMassDeleteModalOpen && null !== attributeGroups && (
+      {isMassDeleteModalOpen && null !== selectedAttributeGroups && (
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:54]: selectedAttributeGroups cannot be null no ?

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx [hunk 4]
```diff
@@ -45,16 +63,41 @@ const ModalMassDeleteAttributeGroups = ({attributeGroups, onConfirm}: MassDelete
         >
           <ModalContent>
             <p>
-              {translate('pim_enrich.entity.attribute_group.mass_delete.confirm', {
-                count: String(attributeGroups.length),
-              })}
+              {translate(
+                'pim_enrich.entity.attribute_group.mass_delete.confirm',
+                {
+                  count: String(selectedAttributeGroups.length),
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:69]: We didn't need to cast it to string
```suggestion
                  count: selectedAttributeGroups.length,
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx [hunk 5]
```diff
@@ -45,16 +63,41 @@ const ModalMassDeleteAttributeGroups = ({attributeGroups, onConfirm}: MassDelete
...
+                  {translate('pim_enrich.entity.attribute_group.mass_delete.attribute_warning', {
+                    number_of_attribute: numberOfAttribute,
+                  })}
+                </Helper>
+                <Field
+                  label={translate('pim_enrich.entity.attribute_group.mass_delete.select_attribute_group', {
+                    number_of_attribute: numberOfAttribute,
+                  })}
+                >
+                  <SelectInput
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:86]: Did we allow to confirm the deletion modal if replacementAttributeGroup is null ?

**Review by StevenVAIDIE — APPROVED** [review]

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx (hunk 5 above)
Comment (MadeWilson) [reply src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:86]: yes in the case you selected only attribute groups that didn't have any attributes.

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx (hunk 1 above)
Comment (MadeWilson) [reply src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/components/datagrids/attribute-groups/ModalMassDeleteAttributeGroups.tsx:36]: Yes I'll wove that where I mount this component

**Pull request merged**